### DISCUSSION TITLE
Any easy way to make the module detect sheets automatically?

### DISCUSSION THREAD

**Discussion by LeCuay** [EddieDover/fvtt-party-sheet#45] [Q&A]
In the template declaration there's `system` property but seems like system is not automatically detected somehow... Any easy way to implement this (meaning changing template name, format, etc...) or is just pure code?

**Comment by EddieDover** [comment]
I'm not exactly sure what the question is here.

The system property is used to determine which sheets to show users on the primary dropdown and only those with systems matching the current game system are displayed. For example, I have about 50 sheets in my partysheets folder, but when I'm in a D&D game ("dnd5e") I only see 4 in the dropdown.

![image](https://github.com/user-attachments/assets/3891bb7c-29e0-4a29-85d1-ef903c142c40)

**Reply by LeCuay** [reply]
Have you tried other systems? Seems to not be working with Pathfinder 1e neither Starfinder

![imagen](https://github.com/user-attachments/assets/c99dc23c-b34f-4a82-8923-3ae01f5b4a44)
![imagen](https://github.com/user-attachments/assets/5bb8b752-353f-49be-b32d-626a1ee5ef5b)

**Reply by EddieDover** [reply]
We test with multiple systems before release. Unfortunately Pathfinder V1 and Starfinder are limited to V11 and we've all moved to V12 at this point, so I can't test those two specifically.

Make sure that the system id matches the actual system id of the system:
![image](https://github.com/user-attachments/assets/1498d61d-8fb7-4282-8922-606098beb9da)
![image](https://github.com/user-attachments/assets/f6e026a9-5dec-4464-943b-60279fb26e17)

Pathfinder's ID is "pf1" and Starfinder is "sfrpg"

**Reply by EddieDover** [reply]
Also I want to note that you can use the console to check and see if your templates are actually seen and loaded:
![image](https://github.com/user-attachments/assets/8df3946d-382c-40c5-a367-0bae468d1f04)

**Comment by LeCuay** [comment]
Question might be poorly worded, basically what am I asking is: _DnD5e_ says to automatically detect the template  (from TEMPLATE_LIST.md)

> - Note: this does not require any upload of a .json, but you can make your own or edit the examples in the [example_templates] folder to your liking and it will show up in the dropdown of available .json files

Can be this done for other systems?

**Reply by EddieDover** [reply]
This is because the module's original design was *only* for dnd5e so I included a built-in template for dnd5e and after the module evolved to encompass any system, I never removed the built-in.  Adding more to the built-in list would just bloat the code and I'd rather not.  In fact, if I had to choose, I'd just remove the built-in template 5e.

**Reply by LeCuay** [reply]
Okay that explains it, thanks! A bummer tho

**Reply by EddieDover** [reply]
The problem with me including more built-in stuff, besides the package size increase,  is that once it's built-in, it's part of the core and becomes my responsibility if it breaks. And, since we've seen that some systems like PF1 and Starfinder refuse to update to new versions in a timely manner, I just can't accept that responsibility.

Honestly, if I had my way, I'd have a separate website or some place that people could upload/edit/etc templates and have the plugin itself read that site and allow users to one-click install/uninstall/preview templates.

It's on my wishlist, but it'll have to wait until I finish the visual editor since every feature we add means that the editor takes even more time to implement properly.

**Reply by LeCuay** [reply]
Item piles handles many different systems and the owner only supports DnD since is the one they play. Everything else is up to the community. How hard could it be to setup something like it?

Actually recently someone from PF1 team fixed support for latest version and support for the major before that
https://github.com/fantasycalendar/FoundryVTT-ItemPiles/pull/644/files#diff-035f25f4f0139c05f3e18806451d24ff0bd613a529fde8b82462ef3c849754c9L49
